framework/functions.php

```diff
@@ -257,14 +257,18 @@ function fw_menu_output ( $menu, $level, $type, $classes ) {
 	echo '">';
 
 	foreach ( $menu as $item ) {
-	
+
 		echo '<li class="';
 
-		if (
-			isset ( $GLOBALS['vars']['current_url'] ) &&
-			$GLOBALS['vars']['current_url'] == $item['url']
-		) {
-			echo 'current-nav-item ';
+		if ( isset ( $GLOBALS['vars']['current_url'] ) ) {
+			$query_position = strpos( $GLOBALS['vars']['current_url'], '?' );
+			$base_current_url = $query_position ?
+				substr( $GLOBALS['vars']['current_url'], 0, $query_position ) :
+				$GLOBALS['vars']['current_url'];
+
+			if ( $base_current_url == $item['url'] ) {
```

Comment:
This logic could probably be more succinct using `parse_url()`:

```php
$url_path = parse_url($GLOBALS['vars']['current_url'], PHP_URL_PATH);
$item_path = parse_url($item['url'], PHP_URL_PATH);

if ( $url_path == $item_path ) {
	echo 'current-nav-item ';
}
```

(**Note:** the logic of line 298 would also need to be updated. Suggestion: since the logic is used in two places, maybe create a variable `$is_item_of_current_page` and use it in those two places).

Reply:
I've modified it a bit to include the HOST in the comparison to make it work with the lang menu on the homepage, when the path is blank.